Patch: 10_wrapper_uid.patch
Author: Tollef Fog Heen <tfheen@debian.org>
Makes sure we're called with the right UID and GID
Index: mailman-2.1.10/src/cgi-wrapper.c
===================================================================
--- mailman-2.1.10.orig/src/cgi-wrapper.c	2008-04-21 20:28:22.000000000 +0200
+++ mailman-2.1.10/src/cgi-wrapper.c	2008-06-15 18:49:41.000000000 +0200
@@ -42,7 +42,7 @@
         char* fake_argv[3];
 
         running_as_cgi = 1;
-        check_caller(logident, parentgroup);
+        if (getgid() >= 100 && getgid() != 65534) check_caller(LOG_IDENT, LEGAL_PARENT_GROUP);
 
         /* For these CGI programs, we can ignore argc and argv since they
          * don't contain anything useful.  `script' will always be the driver
Index: mailman-2.1.10/src/mail-wrapper.c
===================================================================
--- mailman-2.1.10.orig/src/mail-wrapper.c	2008-04-21 20:28:22.000000000 +0200
+++ mailman-2.1.10/src/mail-wrapper.c	2008-06-15 18:49:41.000000000 +0200
@@ -74,7 +74,7 @@
                 fatal(logident, MAIL_ILLEGAL_COMMAND,
                       "Illegal command: %s", argv[1]);
 
-        check_caller(logident, parentgroup);
+       if (getgid() >= 100 && getgid() != 65534) check_caller(logident, LEGAL_PARENT_GROUP);
 
         /* If we got here, everything must be OK */
         status = run_script(argv[1], argc, argv, env);
